packages/website/src/components/layout/EditorTabs.tsx

```diff
@@ -0,0 +1,53 @@
+import EditIcon from '@site/src/icons/edit.svg';
+import clsx from 'clsx';
+import React from 'react';
+
+import styles from './EditorTabs.module.css';
+
+export interface EditorTabsProps<T extends string | boolean> {
+  readonly tabs: ({ value: T; label: string } | T)[];
+  readonly active: T;
+  readonly change: (name: T) => void;
+  readonly showModal?: (name: T) => void;
+}
+
+function EditorTabs<T extends string | boolean>({
+  tabs,
+  active,
+  change,
+  showModal,
+}: EditorTabsProps<T>): JSX.Element {
+  const tabsWithLabels = tabs.map(tab =>
+    typeof tab !== 'object' ? { value: tab, label: String(tab) } : tab,
+  );
+
+  return (
+    <div className={clsx(styles.tabContainer, 'padding--xs')}>
+      <div role="tablist" className="button-group">
+        {tabsWithLabels.map(item => (
+          <button
+            role="tab"
+            className={clsx(styles.tabStyle, 'button')}
+            key={item.label}
+            aria-selected={active === item.value}
+            disabled={active === item.value}
+            onClick={(): void => change(item.value)}
+          >
+            {item.label}
+          </button>
+        ))}
+      </div>
+      {(active === '.eslintrc' || active === 'tsconfig.json') && showModal && (
```

Comment:
Is there ever a case where this check would be necessary? I tried removing it locally and couldn't repro any issues.

```suggestion
      {showModal && (
```

Reply:
this condition hides Visual editor button when you are in `code` tab, 
![image](https://user-images.githubusercontent.com/625469/226490976-567fa972-6aa4-4972-b0e8-57d8a4f9aab8.png)

Reply:
Ahhhh got it!

I'd suggest making this a prop the component takes in, like `hideVisualEditor` or `showVisualEditor`, which is set in the tabs config. Then this component doesn't have to know about the specifics of that quirk.